src/core/src/AssetRegistry.cpp

```diff
@@ -59,6 +62,17 @@ const std::list<std::shared_ptr<OmniImagery>>& AssetRegistry::getAllImageries()
     return _imageries;
 }
 
+const std::list<std::shared_ptr<OmniIonImagery>> AssetRegistry::getAllIonImageries() const {
+    std::list<std::shared_ptr<OmniIonImagery>> ionImageries;
+    for (const auto& imagery : _imageries) {
+        auto ionImagery = std::dynamic_pointer_cast<OmniIonImagery>(imagery);
```

Comment:
I can see that point. I'm a little torn. If we have separate lists for ion imageries, polygon imageries, then WMS and other potential future ones, we'd need to keep them all synced with `AssetRegistry::getAllImageries()` if we wanted to query the whole stage for imageries. Here the dynamic cast is being used as a query filter. The only other way I know to do filtered queries in C++ is by adding something like `OmniImagery::asIonImagery`, but that seems worse.

Would `UsdNotificationHandler` need to handle these differently? If they can be handled the same, then when adding WMS and future imagery classes, we'd have to less near-duplicated code by keeping them in the same vector.

Reply:
I noticed a few bugs in `UsdNotificationHandler` that should be addressed in this PR and could help inform which approach to use. I'm still in favor of separate lists because I think it will reduce the number of areas where dynamic casts or `isA` checks are needed. I think the code duplication is relatively contained. In general I'm just really wary of depending on RTTI.

* `processCesiumImageryAdded` calls `AssetRegistry::addImagery` which creates an `OmniIonImagery`, which is incorrect if it's a polygon imagery. I think the simple fix would be to add `AssetRegistry::addIonImagery`, `AssetRegistry::addPolygonImagery`, etc
* `processCesiumImageryChanged` should be split into `processCesiumIonImageryChanged` and `processCesiumPolygonImageryChanged` since each subclass has different properties that it needs to watch for. To avoid code duplication, they could call a common function that checks for `alpha`, `overlayRenderPipe`, etc

Reply:
Fair point on RTTI. Sure, let's try separate lists.

What `UsdNofiticationHandler` bugs did you notice? This PR didn't do much with that file aside from `s/Imagery/ionImagery/`.

Reply:
Oh, the two bullets above are the bugs.

`UsdNotificationHandler` is still assuming only ion imagery exists. It needs to handle polygon imagery as well. And some things like `overlayRenderPipe` aren't handled, meaning dynamic changes to it won't have any effect.

Reply:
Changed to separate lists. It is nice not to use (much) RTTI. I think the amount of boiler-plate could be reduced more. It sounds like a good, isolated goal to couple with adding WMS layers.

I verified it works:

![image](https://github.com/CesiumGS/cesium-omniverse/assets/1040194/249ec2d1-1b04-4675-aa93-43f8d5c349b3)

Reply:
> Oh, the two bullets above are the bugs.
> 
> `UsdNotificationHandler` is still assuming only ion imagery exists. It needs to handle polygon imagery as well. And some things like `overlayRenderPipe` aren't handled, meaning dynamic changes to it won't have any effect.

@lilleyse It'll handle the two raster-overlay types now. I think the upcoming UI/UX PR is a good time to think through what should be dynamically updated and what not. In Unreal, some things dynamically update, and some require refreshing the tileset.